io-engine/src/bdev/nexus/nexus_bdev.rs

```diff
@@ -99,17 +100,60 @@ impl NvmeAnaState {
     }
 }
 
-/// NVMe-specific parameters for the Nexus
+/// NVMe reservation types.
+#[derive(Debug, Copy, Clone)]
+pub enum NvmeReservation {
+    Reserved = 0,
+    WriteExclusive = 1,
+    ExclusiveAccess = 2,
+    WriteExclusiveRegsOnly = 3,
+    ExclusiveAccessRegsOnly = 4,
+    WriteExclusiveAllRegs = 5,
+    ExclusiveAccessAllRegs = 6,
+}
+impl TryFrom<u8> for NvmeReservation {
+    type Error = ();
+
+    fn try_from(value: u8) -> Result<Self, Self::Error> {
+        Ok(match value {
+            0 => Self::Reserved,
+            1 => Self::WriteExclusive,
+            2 => Self::ExclusiveAccess,
+            3 => Self::WriteExclusiveRegsOnly,
+            4 => Self::ExclusiveAccessRegsOnly,
+            5 => Self::WriteExclusiveAllRegs,
+            6 => Self::ExclusiveAccessAllRegs,
+            _ => return Err(()),
```

Comment:
Shall we return a string error here that provides details ?

Reply:
Yep, we should indeed.

Reply:
Done